### ISSUE TITLE
Make diffs in UI use standard diff colors

### ISSUE THREAD

**Issue by simon-abbott** [vitest-dev/vitest#3108]
### Clear and concise description of the problem

Currently diffs shown in the UI are all red. This makes finding why the test failed more difficult because you can't just scan for the standard red/green differences and find what happened.

### Suggested solution

The diffs shown in the UI should show different colors based on the status of the line (unchanged, added, or removed).

### Alternative

The alternative approach is to just look at the output in the terminal, but that can be hard to parse through, especially if you have a large number of failures (like often happens when doing a large refactor).

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by silverwind** [comment]
"standard diff colors" refers to colors used by `git diff`? 16-color ANSI, right? Generally, I agree, the diff output should color the same as `git`.

**Comment by simon-abbott** [comment]
I don't actually fully care what colors are used, as long as there are three separate colors for unchanged, added, and removed lines. That being said, my preference would be to use the same colors that the CLI output uses (added = green, removed = red, unchanged = default [but not red or green]).

**Comment by hi-ogawa** [comment]
There's an issue of diff color not working even on the terminal by default https://github.com/vitest-dev/vitest/issues/7042

But except that, UI itself now works with color when terminal also shows a color with `FORCE_COLOR=1` for example. 
https://stackblitz.com/edit/vitest-dev-vitest-d8ximaa3?file=package.json

![image](https://github.com/user-attachments/assets/75d702f1-93d7-4c55-97dc-dd57f74c246d)

Let's track the issue in https://github.com/vitest-dev/vitest/issues/7042
